docs/highlighters/regexp.md

```diff
@@ -0,0 +1,24 @@
+zsh-syntax-highlighting / highlighters / regexp
+------------------------------------------------
+
+This is the `regexp` highlighter, that highlights user-defined regular expressions. It's similar to pattern highlighter, but allows more complex patterns.
+
+
+### How to tweak it
+
+To use this highlighter, associate regular expressions with styles in the
+`ZSH_HIGHLIGHT_REGEXP` associative array, for example in `~/.zshrc`:
+
+    ZSH_HIGHLIGHT_REGEXP+=('\bsudo\b' fg=123,bold)
+
+This will highlight "sudo" only if it's surrounded by word boundary, i.e. "sudo cmd", but not "sudoedit"
+
+The syntax for values is the same as the syntax of "types of highlighting" of
+the zsh builtin `$zle_highlight` array, which is documented in [the `zshzle(1)`
+manual page][zshzle-Character-Highlighting].
+
+See also: [regular expressions on wikipedia][regexp-wiki], zsh regexp operator `=~` in [zsh manual page][zsh-man-regexp]
```

Comment:
Not really happy with the link to wikipedia: the article is somewhat on the theoretical side and I suspect not too helpful to those who don't already know the topic.  How about linking to https://docs.python.org/3/howto/regex.html#regex-howto or http://perldoc.perl.org/perlretut.html or http://www.regular-expressions.info/ instead?  Those are more tutorial-y.